src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -24,18 +23,16 @@ public void add(T value) {
     public void add(T value, int index) {
         if (index == size) {
             add(value);
-            return;
         } else {
             checkIndex(index);
-            Node<T> pred = node(index).prev;
-            Node<T> newNode = new Node<>(pred, value, node(index));
-            node(index).prev = newNode;
-            if (pred == null) {
-                first = newNode;
+            if (index == 0) {
+                first.prev = new Node<>(null, value, first);
+                first = first.prev;
             } else {
-                pred.next = newNode;
+                Node<T> newNode = new Node<>(nodeByIndex(index).prev, value, nodeByIndex(index));
```

Comment:
create local variable so you will not need to call method nodeByIndex(index) all the time